anesthetic/plot.py

```diff
@@ -59,314 +145,428 @@ def _constructor(self):
         """Add vertical and horizontal spans across all axes.
 
         Parameters
         ----------
-            params : str or list(str)
-                parameter label(s).
-                Should match the size of `vmins` and `vmaxs`.
-            vmins : float or list(float)
-                Minimum value of the vertical and horizontal axes spans.
-                Should match the size of `params`.
-            vmaxs : float or list(float)
-                Maximum value of the vertical and horizontal axes spans.
-                Should match the size of `params`.
-            kwargs
-                Any kwarg that can be passed to `plt.axvspan` or `plt.axhspan`.
+        params : dict(array_like(2-tuple))
+            Dictionary of parameter labels and desired value tuples.
+            Can provide more than one value tuple per label.
+            Each value tuple provides the min and max value for an axis span.
+        lower, diagonal, upper : bool
+            Whether to plot the spans on the lower, diagonal,
+            and/or upper triangle plots.
+            Default: True
+        kwargs
+            Any kwarg that can be passed to `plt.axvspan` or `plt.axhspan`.
 
         """
         kwargs = normalize_kwargs(kwargs, dict(color=['c']))
-        params = np.ravel(params)
-        vmins = np.ravel(vmins)
-        vmaxs = np.ravel(vmaxs)
-        if params.size != vmins.size:
-            raise ValueError("The sizes of `params`, `vmins` and `vmaxs` must "
-                             "match exactly, but params.size=%s, "
-                             "vmins.size=%s and vmaxs.size=%s."
-                             % (params.size, vmins.size, vmaxs.size))
-        for i, param in enumerate(params):
-            if param in self.columns:
-                for ax in self.loc[:, param]:
-                    if ax is not None:
-                        ax.axvspan(vmins[i], vmaxs[i], **kwargs)
-            if param in self.index:
-                for ax in self.loc[param, self.columns != param]:
-                    if ax is not None:
-                        ax.axhspan(vmins[i], vmaxs[i], **kwargs)
-
-
-def make_1d_axes(params, **kwargs):
+        positions = ['lower' if lower else None,
+                     'diagonal' if diagonal else None,
+                     'upper' if upper else None]
+        for y, rows in self.iterrows():
+            for x, ax in rows.iteritems():
+                if ax is not None and ax.position in positions:
+                    if x in params:
+                        for vmin, vmax in np.atleast_2d(params[x]):
+                            ax.axvspan(vmin, vmax, **kwargs)
+                    if y in params and ax.position != 'diagonal':
+                        for vmin, vmax in np.atleast_2d(params[y]):
+                            ax.axhspan(vmin, vmax, **kwargs)
+
+    def scatter(self, params, lower=True, upper=True, **kwargs):
```

Comment:
Something which came up as part of BeyondPlanck? When is this useful?

Reply:
Yes. Vertical and horizontal lines (which is already implemented with `axlines`) are nice when there are some theoretical values that separate parameter space (e.g. `Omega_K=0` separates open and closed universes). However, when comparing contours to some truth value then it is actually just one n-dimensional point. Emphasis on point, i.e. this doesn't separate parameter space into different regions. So in those situations it would be nicer to highlight the truth with just a point/star/cross/...